Source/Filter/BilinearFilter.cpp

```diff
@@ -18,15 +18,15 @@
 using namespace amrex;
 
 namespace {
-    void compute_stencil(Gpu::DeviceVector<Real> &stencil, int npass)
+    void compute_stencil(Gpu::DeviceVector<Real> &stencil, unsigned int npass)
     {
-        Vector<Real> old_s(1+npass,0.);
-        Vector<Real> new_s(1+npass,0.);
+        Vector<Real> old_s(1u+npass,0.);
+        Vector<Real> new_s(1u+npass,0.);
 
-        old_s[0] = 1.;
+        old_s[0] = 1._rt;
         int jmax = 1;
         // Convolve the filter with itself npass times
-        for(int ipass=1; ipass<npass+1; ipass++){
+        for(int ipass=1; ipass< static_cast<int>(npass+1u); ipass++){
```

Comment:
Avoids an evaluation each loop:
```suggestion
        int const lastpass = static_cast<int>(npass+1u);
        for(int ipass=1; ipass< lastpass; ipass++){
```